### ISSUE TITLE
Wrong syntax color for certain commented words when first loading enh-ruby-mode

### ISSUE THREAD

**Issue by RJSent** [zenspider/enhanced-ruby-mode#170]
If I have a ruby file with a comment containing any of the below words and load that file, the word is highlighted as if it were a method call. (Namely, a method implicity called, like `puts 'Hello'`). The word does not have to be by itself and can have other words around it as long as there's a space between them. The word can also be surrounded in [ ],  ( ), and { }. This happens if I load the file and use enh-ruby-mode automatically, or if I load the file in ruby-mode and switch to enh-ruby-mode with `M-x enh-mo`. 

![enh-ruby-mode-syntax-coloring](https://user-images.githubusercontent.com/25639410/85071860-26a06000-b186-11ea-90c4-a14fd06ba8a3.png)

This issue goes away for an individual line if I edit the line that the word is on. The word is colored like an ordinary comment again. (That's why `__callee__` is green). The issue is also fixed if I reload enh-ruby-mode with `M-x enh-mo` for every occurrence in the file.

I've only observed this with those Kernel methods and not any BasicObject or Object methods.

I tested this with a bare init.el file that only loaded the enh-ruby-mode package using the manual installation code in the readme. (Except for when I loaded the file in ruby-mode and switched to enh-ruby-mode of course)

**Comment by zenspider** [comment]
What's with the squiggle underlines?

**Comment by RJSent** [comment]
That screenshot was when I had flycheck enabled. Here's a new one with a bare init file, except for a manual install of enh-ruby-mode. This one also shows how they can be surrounded with [ ], { }, and ( ).
![enh-ruby-mode-syntax-coloring-2](https://user-images.githubusercontent.com/25639410/85303821-023fce80-b479-11ea-867b-b3939311060b.png)

**Comment by zenspider** [comment]
I can see your behavior cleanly in this way:

```
$ emacs -q -nw --debug-init -l loader.el bug170.rb
```

where loader.el is:

```elisp
(add-to-list 'load-path "." t)
(add-to-list 'auto-mode-alist '("\\.rb$" . enh-ruby-mode))
(setq enh-ruby-program "~/.rubies.current/bin/ruby")
(require 'enh-ruby-mode)
```

**Comment by zenspider** [comment]
I have fixed this bug in f2469f3... but I don't understand the fix at all. The documentation seems to suggest the opposite.

**Comment by RJSent** [comment]
Fixed on my end as well. Thanks!
